### PR TITLE
ENH add na_action to DataFrame.applymap

### LINKED ISSUE #23803: Feature request: add na_action argument to applymap
#### Code Sample, a copy-pastable example if possible

Desired method:

```python
x = pd.DataFrame(...)
x = x.applymap(foo, na_action='ignore')
```

Current method:

```python
x = pd.DataFrame(...)
x = x.apply(lambda y: y.map(foo, na_action='ignore'))
```

#### Problem description

The `na_action='ignore'` argument is extremely useful when processing messy data, and it would be nice to have on the `DataFrame.applymap` method as well.

`DataFrame.applymap` already uses very similar code, so ideally this shouldn't be too dramatic of a change.

[`DataFrame.applymap`](https://github.com/pandas-dev/pandas/blob/v0.23.4/pandas/core/frame.py#L6067-L6072)
[`Series.map` (1)](https://github.com/pandas-dev/pandas/blob/v0.23.4/pandas/core/series.py#L2997-L3000), [`Series.map` (2)](https://github.com/pandas-dev/pandas/blob/v0.23.4/pandas/core/base.py#L994-L1004)

For that matter, is there any reason why `DataFrame.applymap` shouldn't just be a wrapper for `.apply(lambda x: Series.map)`? Seems like it would be beneficial for deduplicating logic in the codebase anyway.

### PR DESCRIPTION
**Pull request by jnothman**
For symmetry with Series.map

- [x] closes #23803
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -7584,7 +7584,7 @@ def apply(self, func, axis=0, raw=False, result_type=None, args=(), **kwds):
         )
         return op.get_result()
 
-    def applymap(self, func) -> "DataFrame":
+    def applymap(self, func, na_action=None) -> "DataFrame":
```
Comment (jreback) [inline pandas/core/frame.py:7587]: can you type the new arg

File: pandas/core/frame.py [hunk 2]
```diff
@@ -7595,6 +7595,10 @@ def applymap(self, func) -> "DataFrame":
         ----------
         func : callable
             Python function, returns a single value from a single value.
+        na_action : {None, 'ignore'}, default None
+            If ‘ignore’, propagate NaN values, without passing them to func.
+
+            .. versionadded:: 1.1
```
Comment (jreback) [inline pandas/core/frame.py:7601]: 1.2

File: pandas/core/frame.py [hunk 3]
```diff
@@ -7633,11 +7646,17 @@ def applymap(self, func) -> "DataFrame":
         0   1.000000   4.494400
         1  11.262736  20.857489
         """
+        if na_action not in {"ignore", None}:
+            raise ValueError(
```
Comment (jreback) [inline pandas/core/frame.py:7650]: is this tested?

File: pandas/tests/frame/apply/test_frame_apply.py [hunk 4]
```diff
@@ -587,6 +587,20 @@ def test_applymap(self, float_frame):
...
+        strlen_frame = float_frame.applymap(lambda x: len(str(x)))
+        float_frame_with_na = float_frame.copy()
+        mask = np.random.randint(0, 2, size=float_frame.shape, dtype=bool)
+        float_frame_with_na[mask] = pd.NA
+        strlen_frame_na_ignore = float_frame_with_na.applymap(
+            lambda x: len(str(x)), na_action="ignore"
+        )
+        strlen_frame_with_na = strlen_frame.copy()
+        strlen_frame_with_na[mask] = pd.NA
+        print(strlen_frame_na_ignore)
```
Comment (jreback) [inline pandas/tests/frame/apply/test_frame_apply.py:600]: no prints!

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jnothman** [comment]
Apologies for my omissions! Thanks for the review.

File: doc/source/whatsnew/v1.2.0.rst [hunk 5]
```diff
@@ -50,6 +50,7 @@ For example:
 
 Other enhancements
 ^^^^^^^^^^^^^^^^^^
+- :meth:`applymap` now supports ``na_action`` (:issue:`23803`)
```
[the remark this answers is not in the document]
Comment (jnothman) [reply doc/source/whatsnew/v1.2.0.rst:53]: I was surprised by similar usage elsewhere in this file, which I copied:
https://github.com/pandas-dev/pandas/blob/6325a33/doc/source/whatsnew/v1.2.0.rst#L35

**Comment by jnothman** [comment]
Failure appears unrelated

**Comment by jreback** [comment]
looks good, can you merge master and ping on green.

**Comment by jnothman** [comment]
@jreback it's green :)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jnothman very nice

**Pull request merged**